whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestBasicAuthTestIntegration.java

```diff
@@ -0,0 +1,273 @@
+
+        assertThat(whoisResources.getErrorMessages(), is(empty()));
+        assertThat(whoisResources.getWhoisObjects(), hasSize(1));
+        final WhoisObject whoisObject = whoisResources.getWhoisObjects().get(0);
+        assertThat(whoisObject.getAttributes(), contains(
+                new Attribute("irt", "irt-test"),
+                new Attribute("address", "RIPE NCC"),
+                new Attribute("e-mail", "noreply@ripe.net"),
+                new Attribute("admin-c", "TP1-TEST", null, "person", Link.create("http://rest-test.db.ripe.net/test/person/TP1-TEST"), null),
+                new Attribute("tech-c", "TP1-TEST", null, "person", Link.create("http://rest-test.db.ripe.net/test/person/TP1-TEST"), null),
+                new Attribute("auth", "MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/", "test", null, null, null),
+                new Attribute("mnt-by", "OWNER-MNT", null, "mntner", Link.create("http://rest-test.db.ripe.net/test/mntner/OWNER-MNT"), null),
+                new Attribute("source", "TEST")));
+    }
+
+    @Test
+    public void create_succeeds_with_basic_auth_no_password_query() {
+        final WhoisResources whoisResources = SecureRestTest.target(getSecurePort(), "whois/test/person")
+                .request()
+                .header("Authorization", getBasicAuthenticationHeader("test", "test"))
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), WhoisResources.class);
+
+        assertThat(whoisResources.getLink().getHref(), is(String.format("https://localhost:%s/test/person",getSecurePort())));
+        assertThat(whoisResources.getErrorMessages(), is(empty()));
+        final WhoisObject object = whoisResources.getWhoisObjects().get(0);
+
+        assertPersonObject(whoisResources, object);
+    }
+
+    @Test
+    public void create_failed_with_basic_auth_incorrect_www_authenticate_cookie() {
+        final Response response = SecureRestTest.target(getSecurePort(), "whois/test/person")
+                .request()
+                .header("Authorization", getBasicAuthenticationHeader("test", "incorrect"))
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), Response.class);
+
+        assertThat(response.getHeaderString(HttpHeaders.WWW_AUTHENTICATE), is(BASIC_CHARSET_ISO_8859_1_LATIN_1));
+    }
+
+    @Test
+    public void create_failed_with_no_basic_auth_incorrect_password_no_www_authenticate_cookie() {
+        final Response response = SecureRestTest.target(getSecurePort(), "whois/test/person?password=incorrect")
+                .request()
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), Response.class);
+
+        assertThat(response.getHeaderString(HttpHeaders.WWW_AUTHENTICATE), is(nullValue()));
+    }
+
+    @Test
+    public void create_failed_with_basic_auth_no_https() {
+
+        final Response response =  RestTest.target(getPort(), "whois/test/person")
```

Comment:
same as before. 
Can we use assertThrows here as well?

Reply:
It is checking the status code if it is not https  as HttpStatus.UPGRADE_REQUIRED_426. DO you mean to check the message inside the response? I will add add that

Reply:
example

        final ForbiddenException throwable = assertThrows(ForbiddenException.class, () ->
            RestTest.target(getPort(), "test/person/TP1-TEST?password=123")
                    .request()
                    .header(HttpHeaders.HOST, getHost(restApiBaseUrl))
                    .header(HttpHeader.X_FORWARDED_PROTO.toString(), HttpScheme.HTTP)
                    .get(WhoisResources.class)
        );

        final Entity entity = throwable.getResponse().readEntity(Entity.class);
        assertThat(entity.getErrorCode(), is(status));

doing this it is clear that there is an exception

Reply:
I think checking status code is perfectly fine too. Status code are unique. For me its more readable and less lines.

Reply:
as you wish, but I prefer to get a solution to apply always. Because in the code I have already seen more than 2 different ways of checking error.
Time ago I talked to Ed about it, and he recommended to use the assertThrows this is why I am doing some refactors time to time in some IT applying this.

Reply:
May be purpose there is to check the exceptions but purpose here is to Response contains upgrade required